test/test.ref

```diff
@@ -0,0 +1,142 @@
+# 1 "test.cppo"
+(* comment *)
+
+# 4 "test.cppo"
+f(1)
+# 6 "test.cppo"
+ 2+ 3.14  
+# 8 "test.cppo"
+ 3+ 3.14  
+
+# 13 "test.cppo"
+"g" is not defined
+
+# 18 "test.cppo"
+ b() 
+
+# 20 "test.cppo"
+debug("a")
+debug("b")
+
+
+
+
+# 33 "test.cppo"
+success
+
+# 45 "test.cppo"
+ 
+"abc\
+ xyz
+ def" 
+(* 123 \
+   789
+   456 *) 
+
+# 48 "test.cppo"
+ "a" "b" 
+
+# 51 "test.cppo"
+ aa ,bb   123   
+
+# 54 "test.cppo"
+  , ) ( 
+
+# 56 "test.cppo"
+blah #define xyz
+
+# 63 "test.cppo"
+  _ _ (* the 2 underscores should be space-separated *) 
+
+# 67 "test.cppo"
+ + + (* there should be some space between the pluses *) 
+
+# 69 "test.cppo"
+(* (* nested comment with single single quote: ' *) "*)" *)
+
+# 72 "test.cppo"
+obj
+   # define 
+# 73 "test.cppo"
+             
+
+# 75 "test.cppo"
+'  (* lone single quote *)
+
+# 78 "test.cppo"
+ 1  
+# 78 "test.cppo"
+    is not 1
```

Comment:
Is this behavior expected?
Context:
https://github.com/ocaml-community/cppo/blob/08b6c59eb53b688d2b66a16c5fa70097918789c0/test/test.cppo#L77-L78

Reply:
This is the output I would expect, yes. Unfortunately, my code from 10+ years ago was not commented in a way that explains what it's meant to achieve :-(

This is a misleading example. I'd change it to an English sentence that makes sense, such as "1 = 1" e.g.
```
#define one 1
one = 1
```

Regarding the expansion: to get a correct source location after replacing some text with a different amount of text, we insert a line directive to avoid shifting the source location. That's why we insert line directives after each expansion. In the original example, `one` is replaced by `1` followed by a line directive on its own line (`# 78 "test.cppo"`), followed by a number of spaces to adjust the position within the line to what it was in the source code. In this case and generally, when expanding into a shorter string that doesn't add newlines, we could do without a line directive, by adding spaces and newlines as needed for padding. Here, `one` would have been replaced by `1` followed by two spaces.

I hope this provides the missing explanations. If not, please ask for further clarifications.